### ISSUE TITLE
Endless loading screens when app enabled

### ISSUE THREAD

**Issue by Jademalo** [derekhe/msfs2020-google-map#50]
Not sure what's causing this, but it's extremely annoying.

if I start MSFS with the google map tool running, the initial loading screen will hang forever. In order to get to the main menu, I need to stop the tool, and only run it once the menu loads.

if I then try to load into a flight with the tool running, it will show some activity in the command prompt window and then eventually stop. The loading screen will then proceed to hang forever.

There are no errors, it just sits on the loading screen indefinitely.

I'm currently trying to locally cache the area I'm wanting to fly, to see if that will improve things. At present though, I cannot get ingame with the tool running.

**Comment by ganeshh123** [comment]
Same issue with me on the main loading screen, but loading into the flight works okay

**Comment by derekhe** [comment]
I think it is a DNS issue, I'm trying to find out.

**Comment by derekhe** [comment]
Please try this version: https://github.com/derekhe/msfs2020-google-map/releases/tag/2.2.5

**Comment by ganeshh123** [comment]
Sorry, the newest version doesn't even start for me:
![image](https://user-images.githubusercontent.com/43777439/141521723-3591509a-cf97-4484-b97c-c991b63cf116.png)
It's not finding the `nginx` executable, here is the full log:
```

C:\Games\Microsoft Flight Simulator\google-maps-data>if exist .\Python39\python.exe (
echo "Use embedded python"
 .\Python39\python.exe src\app.py
)  else (
echo "Use user python"
 py -m pip install -r requirements.txt
 py src\app.py
)
"Use embedded python"
Restoring hosts
ERROR: The process "nginx.exe" not found.
Restoring hosts
root "Trusted Root Certification Authorities"
Related Certificates:

Exact match:
Element 2:
Serial Number: c9dfffaa7d7e8ef1c6bc38e6781f0fb1
Issuer: CN=mkcert DESKTOP-O2CB4QF\derekhe@DESKTOP-O2CB4QF, OU=DESKTOP-O2CB4QF\derekhe@DESKTOP-O2CB4QF, O=mkcert development CA
 NotBefore: 8/18/2021 14:01
 NotAfter: 11/18/2023 14:01
Subject: OU=DESKTOP-O2CB4QF\derekhe@DESKTOP-O2CB4QF, O=mkcert development certificate
Non-root Certificate
Cert Hash(sha1): ae4cfdbea30eaa635443b6914b3b863b52f693de

Certificate "DESKTOP-O2CB4QF\derekhe@DESKTOP-O2CB4QF" already in store.
CertUtil: -addstore command completed successfully.
Traceback (most recent call last):
  File "C:\Games\Microsoft Flight Simulator\google-maps-data\src\app.py", line 249, in run
    output = self.config_dns(output)
  File "C:\Games\Microsoft Flight Simulator\google-maps-data\src\app.py", line 225, in config_dns
    for k, v in get_hosts_origin_ips().items():
  File "C:\Games\Microsoft Flight Simulator\google-maps-data\src\runner.py", line 21, in get_hosts_origin_ips
    origin_ips[d] = dns_resolver.resolve(d)[0].to_text()
  File "C:\Games\Microsoft Flight Simulator\google-maps-data\Python39\lib\site-packages\dns\resolver.py", line 1176, in resolve
    timeout = self._compute_timeout(start, lifetime)
  File "C:\Games\Microsoft Flight Simulator\google-maps-data\Python39\lib\site-packages\dns\resolver.py", line 997, in _compute_timeout
    raise Timeout(timeout=duration)
dns.exception.Timeout: The DNS operation timed out after 5.4150450229644775 seconds

```

**Comment by mattcleeds** [comment]
> Not sure what's causing this, but it's extremely annoying.
> 
> if I start MSFS with the google map tool running, the initial loading screen will hang forever. In order to get to the main menu, I need to stop the tool, and only run it once the menu loads.
> 
> if I then try to load into a flight with the tool running, it will show some activity in the command prompt window and then eventually stop. The loading screen will then proceed to hang forever.
> 
> There are no errors, it just sits on the loading screen indefinitely.
> 
> I'm currently trying to locally cache the area I'm wanting to fly, to see if that will improve things. At present though, I cannot get ingame with the tool running.

For me this was fixed by: 

1. loading MSFS without starting the mod
2. in MSFS going to general settings, and Data 
3. scroll to the bottom and clear the rolling cache. (You can switch the cache straight back on if you like) 
4. then close MSFS 
5. start the mod 
6. reload MSFS.

hope this helps

**Comment by Jademalo** [comment]
This was actually exactly what I did, but even then I could not get past the initial screen.
I'll try the new version now and report back.

**Comment by derekhe** [comment]
@Jademalo please check if you have these two lines in your hosts file:
* 127.0.0.1 kh.ssl.ak.tiles.virtualearth.net
* 127.0.0.1 khstorelive.azureedge.net
If yes, remove them before you start MSFS then you should get into main page.
Then start the mod.

**Comment by derekhe** [comment]
@Jademalo fixed or I will close this issue.

**Comment by Jademalo** [comment]
Apologies, haven't had much spare time over the last two weeks, been extremely busy. 

Had a quick 5 minutes to test just now. uHosts is empty, however same issue still persists. If the server is started before the game is opened, then the game will hang on the loading screen. It hangs as soon as the images appear that show the newest features in the most recent patch, and will just sit there indefinitely.

**Comment by derekhe** [comment]
Please try latest version: https://github.com/derekhe/msfs2020-google-map/releases/tag/2.3.2

**Comment by Jademalo** [comment]
Good news, that seems to have fixed it!
Running the latest version I was able to successfully get past the loading screen and in game without too many issues.

The first time I tried to load I had this error;
![image](https://user-images.githubusercontent.com/386846/144704877-ce8f3b4a-845d-4706-af96-f563e945f60c.png)
though after loading the game without the mod, closing it, starting the mod, and loading the game again, everything seemed to work.

Sorry for taking so long to test, always busy around this time of year.
